server/server.go

```diff
@@ -887,3 +876,25 @@ func sortedFrontendNamesForConfig(configuration *types.Configuration) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+func (server *Server) configureFrontends(frontends map[string]*types.Frontend) {
+	for _, frontend := range frontends {
+		// default endpoints if not defined in frontends
+		if len(frontend.EntryPoints) == 0 {
+			frontend.EntryPoints = server.globalConfiguration.DefaultEntryPoints
+		}
+	}
+}
+
+func configureBackends(backends map[string]*types.Backend) {
```

Comment:
@timoreimann I think there is at least 2 reasons to add `func (server *Server)` to both functions:

1. it's homogeneous (to the reader, it would be easier to understand if `configureFrontends` and `configureBackends` add the same signature)
2. we could only call `configureBackends` if we had a reference to a `Server` which is makes more sense